ci-operator/step-registry/cucushift/upgrade/toimage/cucushift-upgrade-toimage-commands.sh

```diff
@@ -0,0 +1,201 @@
+        echo "Check on oc adm upgrade --to-image missing --allow-explicit-upgrade PASSED"
+    else
+        echo >&2 "Check on oc adm upgrade --to-image missing --allow-explicit-upgrade FAILED: $err, exiting" && exit 1
+    fi
+    local current_image; current_image="$(./oc get clusterversion version -o json|jq -r '.status.history[0].image')"
+    [[ $1 == "${current_image}" ]] && echo >&2 "Upgrade with oc adm upgrade --to-image missing --allow-explicit-upgrade starts but expected NOT start, exiting" && exit 1
+    
+    if [[ "${FORCE_UPDATE}" == "false" ]]; then
+        echo "Upgrading cluster to $1..."
+        if ! ./oc adm upgrade --to-image="$1" --allow-explicit-upgrade; then
+            echo >&2 "Failed to start upgrade, exiting" && exit 1
+        fi
+    else
+        echo "Force upgrading cluster to $2..."
+        if ! ./oc adm upgrade --to-image="$2" --allow-explicit-upgrade --force; then
+            echo >&2 "Failed to start upgrade, exiting" && exit 1
+        fi
+    fi
+
+    set +x
+}
+
+# Monitor the upgrade status
+# Parameter: 1-target release version
+function check_upgrade_status() {
+    local timeout=120
+    local out="", avail="", progress=""
+    while [ ${timeout} -gt 0 ]; do
+        echo "./oc get clusterversion" && ./oc get clusterversion
+        out="$(./oc get clusterversion --no-headers)"
+        avail="$(echo "${out}" | awk '{print $3}')"
+        progress="$(echo "${out}" | awk '{print $4}')"
+        if [[ "${avail}" == "True" && "${progress}" == "False" && "${out}" == *"Cluster version is $1" ]]; then
+            echo -e "Upgrade succ\n\n"
+            break
+        else
+            sleep 5m
+            ((timeout-=5))
+        fi        
+    done
+    # Collect co, node, mcp, cv state for further debug
+    echo "./oc get co" && ./oc get co
+    echo "./oc get node" && ./oc get node
+    echo "./oc get mcp" && ./oc get mcp
+    echo "./oc get clusterversion version -oyaml" && ./oc get clusterversion version -oyaml
+    [ "${timeout}" -le 0 ] && echo >&2 "Upgrade timeout, exiting" && exit 1
+
+}
+
+# Check version, state, verified in history
+# Parameter: 1- release version
+function check_history() {
+    local history; history=$(./oc get clusterversion/version -o json)
+    local version; version=$(echo "${history}" | jq -r '.status.history[0].version')
+    local state; state=$(echo "${history}" | jq -r '.status.history[0].state')
+    local verified; verified=$(echo "${history}" | jq -r '.status.history[0].verified')
+    if [[ "${version}" == "$1" && "${state}" == "Completed" ]]; then
+        echo "History check PASSED, cluster is now upgraded to $1"
+    else
+        echo >&2 "History check FAILED, cluster upgrade to $1 failed, current version is ${version}, exiting" && exit 1
+    fi
+    if [[ "${FORCE_UPDATE}" == "false" ]]; then
+        if [[ "${signed}" == "True" && "${verified}" == "true" || "${signed}" == "False" && "${verified}" == "false" ]]; then
```

Comment:
Can we unify the return value? `True` -> `true` and `False`-> `false' for readability?